keyboards/ergodox_extended/matrix.c

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2013 Oleg Kostyuk <cub.uanic@gmail.com>
+Copyright 2017 Erin Call <hello@erincall.com>
+Copyright 2024 Adam Coleman <gephorian@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <stdint.h>
+#include <stdbool.h>
+#include <avr/io.h>
+#include "wait.h"
+#include "print.h"
+#include "util.h"
+#include "matrix.h"
+#include "i2c_master.h"
+
+void init_expander(void);
+
+static const uint8_t onboard_row_pins[MATRIX_ROWS] = MATRIX_ONBOARD_ROW_PINS;
+static const uint8_t onboard_col_pins[MATRIX_COLS] = MATRIX_ONBOARD_COL_PINS;
+static const bool col_expanded[MATRIX_COLS] = COL_EXPANDED;
+
+/* matrix state(1:on, 0:off) */
+static matrix_row_t matrix[MATRIX_ROWS];
```

Comment:
```suggestion
```

Reply:
It won't compile without this. Any includes I'm missing or is it fine to leave it?

Reply:
Compiles fine for me
```diff
diff --git a/keyboards/ergodox_extended/matrix.c b/keyboards/ergodox_extended/matrix.c
index 76c4013c12..b208ebb0df 100644
--- a/keyboards/ergodox_extended/matrix.c
+++ b/keyboards/ergodox_extended/matrix.c
@@ -31,9 +31,6 @@ static const uint8_t onboard_row_pins[MATRIX_ROWS] = MATRIX_ONBOARD_ROW_PINS;
 static const uint8_t onboard_col_pins[MATRIX_COLS] = MATRIX_ONBOARD_COL_PINS;
 static const bool col_expanded[MATRIX_COLS] = COL_EXPANDED;
 
-/* matrix state(1:on, 0:off) */
-static matrix_row_t matrix[MATRIX_ROWS];
-
 static const uint8_t expander_col_pins[MATRIX_COLS] = MATRIX_EXPANDER_COL_PINS;
 static void init_cols(void);
 static bool read_cols_on_row(matrix_row_t current_matrix[], uint8_t current_row);
@@ -52,11 +49,6 @@ void matrix_init_custom(void)
 
     unselect_rows();
     init_cols();
-
-    // initialize matrix state: all keys off
-    for (uint8_t i=0; i < MATRIX_ROWS; i++) {
-        matrix[i] = 0;
-    }
 }
 
 void init_expander(void) {
 ```

```
Checking file size of ergodox_extended_default.hex                                                  [OK]
 * The firmware size is fine - 17652/28672 (61%, 11020 bytes free)
```

Reply:
I'm still learning. Thank you for your patience.